enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -8160,18 +8239,25 @@ class AdjointGenerator
           getForwardBuilder(Builder2);
 
           SmallVector<Value *, 2> args;
-#if LLVM_VERSION_MAJOR >= 14
-          for (unsigned i = 0; i < orig->arg_size(); ++i)
-#else
-          for (unsigned i = 0; i < orig->getNumArgOperands(); ++i)
-#endif
-          {
+
+          for (unsigned i = 0; i < orig->getNumArgOperands(); ++i) {
             auto arg = orig->getArgOperand(i);
             args.push_back(gutils->getNewFromOriginal(arg));
           }
-          CallInst *CI = Builder2.CreateCall(orig->getFunctionType(),
-                                             orig->getCalledFunction(), args);
-          CI->setAttributes(orig->getAttributes());
+
+          Value *CI;
+          if (gutils->getWidth() > 1) {
+            CI = applyChainRule(call.getType(), Builder2, [&]() {
```

Comment:
Can this be done without the width check?

Reply:
yes